autogpt_platform/frontend/src/app/signup/actions.ts

```diff
@@ -0,0 +1,44 @@
+"use server";
+import { revalidatePath } from "next/cache";
+import { redirect } from "next/navigation";
+import { z } from "zod";
+import * as Sentry from "@sentry/nextjs";
+import getServerSupabase from "@/lib/supabase/getServerSupabase";
+import { signupFormSchema } from "@/types/auth";
+
+export async function signup(values: z.infer<typeof signupFormSchema>) {
+  "use server";
+  return await Sentry.withServerActionInstrumentation(
+    "signup",
+    {},
+    async () => {
+      const supabase = getServerSupabase();
+
+      if (!supabase) {
+        redirect("/error");
+      }
+
+      // We are sure that the values are of the correct type because zod validates the form
+      const { data, error } = await supabase.auth.signUp(values);
+
+      if (error) {
+        console.error("Error signing up", error);
+        // FIXME: supabase doesn't return the correct error message for this case
```

Comment:
I couldn't make supabase return actual error on database trigger, it returns `unexpected_failure` instead, so any error triggers waitlist prompt in `signup/page.tsx`.

Reply:
Shouldn't we propagate the error either way or say something like "Internal/unknown error" ?

Reply:
I tried fixing it but couldn't, supabase returns a generic database error instead of the custom one from the function (set in the supabase dashboard). This is just to show waitlist information (see screenshot).